openedx/core/lib/xblock_pipeline/finder.py

```diff
@@ -109,13 +111,30 @@ class XBlockPipelineFinder(BaseFinder):
     A static files finder that gets static assets from xblocks.
     """
     def __init__(self, *args, **kwargs):
+        """
+        XBlock class assets are ultimately accessed via pkg_resources. However,
+        if multiple XBlocks are installed by the same setup.py, they will share
+        the same resource files. Instead of writing out copies of those files
+        for every XBlock, we collapse them by top level package. So for instance,
+        "problem_builder.answer.AnswerBlock" and
+        "problem_builder.choice.ChoiceBlock" are both installed by the same
+        setup.py file and store their assets in a folder for "problem_builder".
+
+        However, XBlocks can change how they reference their resources, using a
+        different prefix. It's probably not a great idea to change this value
+        between XBlocks in the same install package, but it is possible, so we
+        iterate through all unique combinations of (package, prefix).
```

Comment:
Question: I don't see "prefix" in the code here?  However, I do see the call to `get_resources_dir` on the `xblock_class`, which is included in the set of tuples.  Is the "prefix" that is referred to here used in the `get_resources_dir` logic?

Reply:
Yeah. I'll rewrite the comments to be clearer. Thank you.